nssp_demo/timeseries_forecasts.R

```diff
@@ -85,35 +154,100 @@ main <- function(model_run_dir, n_forecast_days = 28, n_samples = 2000) {
       ed_visits_target = Disease, ed_visits_other = Other,
       data_type
-    ) |>
-    as_tsibble(index = date)
-
-  forecast_other <- fit_and_forecast(target_and_other_data, n_forecast_days,
+    )
+  ## Time series forecasting
+  ## Fit and forecast other (non-target-disease) ED visits using a combination
+  ## ensemble model
+  forecast_other <- fit_and_forecast(
+    target_and_other_data,
+    n_forecast_days,
     n_samples,
-    target_col = "ed_visits_other", output_col = "other_ed_visits"
+    target_col = "ed_visits_other",
+    output_col = "other_ed_visits"
   )
-  forecast_baseline <- fit_and_forecast(target_and_other_data, n_forecast_days,
+  forecast_baseline_ts_count <- fit_and_forecast(
+    target_and_other_data,
+    n_forecast_days,
     n_samples,
     target_col = "ed_visits_target",
-    output_col = "baseline_ed_visits"
+    output_col = "baseline_ed_visit_count_forecast"
+  )
+  ## Generate CDC flat forecast for the target disease number of ED visits
+  forecast_baseline_cdc_count <- cdc_flat_forecast(
+    target_and_other_data,
+    target_col = "ed_visits_target",
+    output_col = "baseline_ed_visit_count_forecast",
+    data_frequency = "1 day",
+    aheads = 1:n_forecast_days
+  )
+
+  forecast_baseline_ts_prop <- forecast_baseline_ts_count |>
+    to_prop_forecast(forecast_other)
+
+  forecast_baseline_cdc_prop <- cdc_flat_forecast(
+    target_and_other_data |>
+      mutate(ed_visits_prop = ed_visits_target /
+        (ed_visits_target + ed_visits_other)),
+    target_col = "ed_visits_prop",
+    output_col = "baseline_ed_visit_prop_forecast",
+    data_frequency = "1 day",
+    aheads = 1:n_forecast_days
   )
 
-  save_path_other <- path(model_run_dir, "other_ed_visits_forecast",
+  save_path_other <- path(
+    model_run_dir,
+    "other_ed_visits_forecast",
+    ext = "parquet"
+  )
+  save_path_baseline_ts_count <- path(
+    model_run_dir,
+    "baseline_ts_count_ed_visits_forecast",
+    ext = "parquet"
+  )
+  save_path_baseline_ts_prop <- path(
+    model_run_dir,
+    "baseline_ts_prop_ed_visits_forecast",
     ext = "parquet"
   )
-  save_path_baseline <- path(model_run_dir, "baseline_ed_visits_forecast",
+  save_path_baseline_cdc_count <- path(
+    model_run_dir,
+    "baseline_cdc_count_ed_visits_forecast",
     ext = "parquet"
   )
-  write_parquet(forecast_other, save_path_other)
-  write_parquet(forecast_baseline, save_path_baseline)
+  save_path_baseline_cdc_prop <- path(
+    model_run_dir,
+    "baseline_cdc_prop_ed_visits_forecast",
+    ext = "parquet"
+  )
+
+  write_parquet(
+    forecast_other,
+    save_path_other
+  )
+  write_parquet(
+    forecast_baseline_ts_count,
+    save_path_baseline_ts_count
+  )
+  write_parquet(
+    forecast_baseline_ts_prop,
+    save_path_baseline_ts_prop
+  )
+  write_parquet(
+    forecast_baseline_cdc_count,
+    save_path_baseline_cdc_count
+  )
+  write_parquet(
+    forecast_baseline_cdc_prop,
+    save_path_baseline_cdc_prop
+  )
```

Comment:
`walk2` instead of so much repeated code?

Reply:
And/or split out the baseline and the timeseries forecaster

Reply:
Done in f1cda8f